common/raw_hashtable.h

```diff
@@ -494,93 +577,103 @@ inline auto ComputeSeed() -> uint64_t {
 inline auto ComputeProbeMaskFromSize(ssize_t size) -> size_t {
   CARBON_DCHECK(llvm::isPowerOf2_64(size))
       << "Size must be a power of two for a hashed buffer!";
-  // The probe mask needs to mask down to keep the index within
-  // `size`. Since `size` is a power of two, this is equivalent to
-  // `size - 1`. We also mask off the low bits while here to match the size of
-  // the groups of entries.
+  // Since `size` is a power of two, we can make sure the probes are less
+  // than `size` by making the mask `size - 1`. We also mask off the low
+  // bits so the probes are a multiple of the size of the groups of entries.
   return (size - 1) & ~GroupMask;
 }
 
 // This class handles building a sequence of probe indices from a given
 // starting point, including both the quadratic growth and masking the index
 // to stay within the bucket array size. The starting point doesn't need to be
-// clamped to the size ahead of time (or even by positive), we will do it
+// clamped to the size ahead of time (or even be positive), we will do it
 // internally.
 //
+// For reference on quadratic probing:
+// https://en.wikipedia.org/wiki/Quadratic_probing
+//
 // We compute the quadratic probe index incrementally, but we can also compute
 // it mathematically and will check that the incremental result matches our
 // mathematical expectation. We use the quadratic probing formula of:
 //
-//   p(x,s) = (x + (s + s^2) / 2) mod (Size / GroupSize)
-//
-// This particular quadratic sequence will visit every value modulo the
-// provided size divided by the group size.
+//   p(start, step) = (start + (step + step^2) / 2) (mod size / GroupSize)
 //
-// However, we compute it scaled to the group size constant G and have it visit
-// each G multiple modulo the size using the scaled formula:
-//
-//   p(x,s) = (x + (s + (s^2 * G) / G^2) / 2) mod Size
+// However, we compute it incrementally and scaled to the group size. We scale
+// everything back down in order to validate that we are in fact correctly
+// following the above formula.
```

Comment:
The scaling back down bit is an implementation detail that doesn't need to be surfaced at this level and is already addressed below. Instead, it is better to give the motivation / why for the scaling.

```suggestion
// However, we compute it incrementally and scale all the variables by the group size
// so it can be used as an index without an additional multiplication.
```